### PR TITLE
ip-reconciler: do not reconcile pods in Pending phase

### PR DESCRIPTION
**Pull request by maiqueb**
Fixes: #162 

Would you review / check if this works for you @xagent003 ?

### CODE REVIEW

File: cmd/reconciler/ip_test.go [hunk 1]
```diff
@@ -252,6 +252,48 @@ var _ = Describe("Whereabouts IP reconciler", func() {
...
+				metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		BeforeEach(func() {
+			pool = generateIPPoolSpec(ipRange, namespace, poolName, pod.Name)
+			Expect(k8sClient.Create(context.Background(), pool)).NotTo(HaveOccurred())
+			Expect(pool.Spec.Allocations).To(ConsistOf(v1alpha1.IPAllocation{
+				PodRef: fmt.Sprintf("%s/%s", namespace, podName),
+			}))
```
Comment (maiqueb) [inline cmd/reconciler/ip_test.go:276]: I probably should just delete this part. This is testing the testing function ...

**Comment by xagent003** [comment]
@maiqueb I tested with the slight change to logic as above, it is working. I added a temporary log in the new if clause to verify it is being hit. You can see the JSON parsing fails earlier due to missing network status, then we hit the new condition for Pending phase, and the IP is not removed.

```
2021-11-16T23:06:02Z [debug] NewReconcileLooper - Kubernetes config file located at: /host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig
2021-11-16T23:06:03Z [debug] successfully read the kubernetes configuration file located at: /host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig
2021-11-16T23:06:04Z [debug] listing IP pools
2021-11-16T23:06:04Z [debug] Added IP 10.128.165.32 for pod default/web-0
2021-11-16T23:06:04Z [debug] Added IP 10.128.40.45 for pod default/web-0
2021-11-16T23:06:04Z [error] could not parse network annotation  for pod: default/web-1; error: unexpected end of JSON input
2021-11-16T23:06:04Z [debug] the IP reservation: IP: 10.128.165.33 is reserved for pod: default/web-1
2021-11-16T23:06:04Z [debug] PF9: Skipping Pod IP check for Pending Pod: default/web-1 IP: 10.128.165.33
2021-11-16T23:06:04Z [debug] the IP reservation: IP: 10.128.165.32 is reserved for pod: default/web-0
2021-11-16T23:06:04Z [debug] pod reference default/web-0 matches allocation; Allocation IP: 10.128.165.32; PodIPs: map[10.128.165.32:{} 10.128.40.45:{}]
2021-11-16T23:06:04Z [debug] the IP reservation: IP: 10.128.40.45 is reserved for pod: default/web-0
2021-11-16T23:06:04Z [debug] pod reference default/web-0 matches allocation; Allocation IP: 10.128.40.45; PodIPs: map[10.128.165.32:{} 10.128.40.45:{}]
2021-11-16T23:06:04Z [debug] the IP reservation: IP: 10.128.40.46 is reserved for pod: default/web-1
2021-11-16T23:06:04Z [debug] PF9: Skipping Pod IP check for Pending Pod: default/web-1 IP: 10.128.40.46
2021-11-16T23:06:04Z [debug] pod reference default/web-0 matches allocation; Allocation IP: 10.128.165.32; PodIPs: map[10.128.165.32:{} 10.128.40.45:{}]
2021-11-16T23:06:04Z [debug] PF9: Skipping Pod IP check for Pending Pod: default/web-1 IP: 10.128.165.33
2021-11-16T23:06:04Z [debug] pod reference default/web-0 matches allocation; Allocation IP: 10.128.40.45; PodIPs: map[10.128.165.32:{} 10.128.40.45:{}]
2021-11-16T23:06:04Z [debug] PF9: Skipping Pod IP check for Pending Pod: default/web-1 IP: 10.128.40.46
2021-11-16T23:06:04Z [debug] no IP addresses to cleanup

```

File: cmd/reconciler/ip_test.go (hunk 1 above)
Comment (maiqueb) [reply cmd/reconciler/ip_test.go]: Done.

**Review by dougbtv — APPROVED** [review]
LGTM! Thanks for the patch Miguel and the eyes Arjun.

**Pull request merged**